From 9bb59f922da4474e3b30e7f47cf841c9e40c3ffd Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Thu, 18 Dec 2014 00:52:48 +0100
Subject: [PATCH] dispatcher: readme file updated

---
 modules/dispatcher/README | 38 +++++++++++++++++++++++++++++++++++---
 1 file changed, 35 insertions(+), 3 deletions(-)

diff --git a/modules/dispatcher/README b/modules/dispatcher/README
index 1987823..8f2c96a 100644
--- a/modules/dispatcher/README
+++ b/modules/dispatcher/README
@@ -91,6 +91,10 @@ Carsten Bock
         7. Installation and Running
 
               7.1. Destination List File
+
+                    7.1.1. Special Attributes
+                    7.1.2. File Format
+
               7.2. Kamailio config file
 
         8. Event routes
@@ -207,6 +211,10 @@ Chapter 1. Admin Guide
    7. Installation and Running
 
         7.1. Destination List File
+
+              7.1.1. Special Attributes
+              7.1.2. File Format
+
         7.2. Kamailio config file
 
    8. Event routes
@@ -1039,6 +1047,10 @@ kamcmd dispatcher.set_state ip 2 sip:127.0.0.1:5080
 7. Installation and Running
 
    7.1. Destination List File
+
+        7.1.1. Special Attributes
+        7.1.2. File Format
+
    7.2. Kamailio config file
 
 7.1. Destination List File
@@ -1056,9 +1068,29 @@ kamcmd dispatcher.set_state ip 2 sip:127.0.0.1:5080
        probing destination (sending keep alives);
      * priority: sets the priority in destination list (based on it is
        done the initial ordering inside the set)
-     * attributes: extra filed in form of name1=value1;...;nameN=valueN.
-       There are some predefined names that are used of weight and call
-       load dispatching.
+     * attributes: extra fields in form of name1=value1;...;nameN=valueN.
+       There are some reserved names for attributes used by the module for
+       specific load balancing algorithms, see the section 'Special
+       Attributes'.
+
+7.1.1. Special Attributes
+
+   There are some predefined names:
+     * 'duid' - used for call load dispatching. It must be an unique value
+       to identify a destination (gateway address). Practically the load
+       within the group is associated with this value.
+     * 'maxload' - used for call load dispatching. It must be a positive
+       integer, defining the upper limit of active calls per destination.
+       When the limit is reached, then the gateway is no longer selected
+       for new calls until an exiting call via that gateway is terminated.
+       If set to 0, then no active call limit is used.
+     * 'weight' - used for weight based load distribution. It must be set
+       to a positive integer value beteen 0 and 100. The value represents
+       the percent of calls to be sent to that gateways.
+     * 'socket' - used to set the sending socket for the gateway. It is
+       used for sending the SIP traffic as well as OPTIONS keepalives.
+
+7.1.2. File Format
 
    Line format is:
 ...
-- 
2.1.4

