From 11dc171ae98212bb22321e03f3c10e15b6017a80 Mon Sep 17 00:00:00 2001
From: lazedo <luis.azedo@factorlusitano.com>
Date: Fri, 26 Dec 2014 18:46:41 +0000
Subject: [PATCH] kazoo : fork process with tcp option

---
 modules/kazoo/kazoo.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/kazoo/kazoo.c b/modules/kazoo/kazoo.c
index 2f96e8a..466919d 100644
--- a/modules/kazoo/kazoo.c
+++ b/modules/kazoo/kazoo.c
@@ -276,7 +276,7 @@ static int mod_child_init(int rank)
 
 
 	if (rank==PROC_MAIN) {
-		pid=fork_process(1, "AMQP Manager", 0);
+		pid=fork_process(1, "AMQP Manager", 1);
 		if (pid<0)
 			return -1; /* error */
 		if(pid==0){
@@ -284,7 +284,7 @@ static int mod_child_init(int rank)
 		}
 		else {
 			for(i=0; i < dbk_consumer_processes; i++) {
-				pid=fork_process(i+2, "AMQP Consumer", 0);
+				pid=fork_process(i+2, "AMQP Consumer", 1);
 				if (pid<0)
 					return -1; /* error */
 				if(pid==0){
-- 
2.1.4

